File: artifacts/mimalloc/decompose/src/mi_block_visit_fun.c
typedef bool mi_block_visit_fun(const mi_heap_t *heap, const mi_heap_area_t *area, void *block, size_t block_size, void *arg)
/*** DEPENDENCIES:
typedef struct mi_heap_area_s
{
  void *blocks;
  size_t reserved;
  size_t committed;
  size_t used;
  size_t block_size;
  size_t full_block_size;
  int heap_tag;
} mi_heap_area_t
----------------------------
None
----------------------------
***/

File: artifacts/mimalloc/manual_debug/src/mi_block_visit_fun.rs
use crate::*;
use std::sync::atomic::AtomicUsize;
use std::sync::atomic::Ordering;


pub type MiBlockVisitFun = fn(
    heap: Option<&MiHeapS>,
    area: Option<&MiHeapAreaS>,
    block: Option<&[u8]>,
    block_size: usize,
    arg: Option<&[u8]>,
) -> bool;

